api-reference/10 UI Components/dxScheduler/1 Configuration/appointmentTooltipTemplate.md

```diff
@@ -29,6 +29,399 @@ A template name or container.
+                    @click="onClick"
+                ></DxButton>
+            </div>
+        </div>
+    </template>
+
+    <script>
+    import {DxButton} from 'devextreme-vue/button';
+    import { employees } from './data.js';
+
+    export default {
+        components: {
+            DxButton,
+        },
+        data() {
+            return {
+                employees,
+            };
+        },
+        methods: {
+            onClick(e){
+                this.$emit('delete-button-click', e);
+            },
+        },
+        props: {
+            data: {
+                type: Object,
+                default: () => {},
+            },
+            markerColor: {
+                type: String
+            },
+            isDeleteButtonExist: {
+                type: Boolean
+            }
+        },
+    };
+    </script>
+
+
+##### React
+
+    <!-- tab: App.js -->
+    import React, { useCallback } from 'react';
+
+    import 'devextreme/dist/css/dx.light.css';
+
+    import Scheduler from 'devextreme-react/scheduler';
+
+    import Tooltip from "./Tooltip.js";
+
+    const App = () => {
+        const schedulerRef =  useRef(null);
+
+        const appointmentTooltip = useCallback((props) => {
+            const scheduler = schedulerRef.current;
+
+            const getDisabled = (employeeID) => employees.find((employee) => {
+                return employee.id === employeeID;
+            }).disabled;
+
+            const getColor = (employeeID) => employees.find((employee) => {
+                return employee.id === employeeID;
+            }).color;
+
+            const color = getColor(props.appointmentData.employeeID);
+
+            const isDeleteButtonExist = !getDisabled(props.appointmentData.employeeID) &&
+                ((scheduler.instance.option("editing") &&
+                    scheduler.instance.option("editing.allowDeleting") === true) ||
+                    scheduler.instance.option("editing") === true);
+
+            const onDeleteButtonClick = (e) => {
+                scheduler.instance.deleteAppointment(props.appointmentData);
+                e.event.stopPropagation();
+                scheduler.instance.hideAppointmentTooltip();
+            };
+
+            return (
+                <Tooltip
+                    {...props}
+                    isDeleteButtonExist={isDeleteButtonExist}
+                    onDeleteButtonClick={onDeleteButtonClick}
+                    color={color}
+                />
+            );
+        }, []);
+
+        render(
+            <Scheduler ...
+                ref={schedulerRef}
+                appointmentTooltipRender={appointmentTooltip}
+            >
+            </Scheduler>
+        );
+    }
+
+    export default App;
+
+    <!-- tab: Tooltip.js -->
+    import React, { useCallback } from 'react';
+
+    const  Tooltip = (props) => {
+        const onDeleteButtonClick = useCallback((e) => {
+            props.onDeleteButtonClick(e);
+        }, []);
```

Comment:
We should mention the ```props.onDeleteButtonClick``` in the ```useCallback``` dependencies array:

```suggestion
        }, [props.onDeleteButtonClick]);
```

Otherwise, if we change the function in the props, the old one still be called on click :)